components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java

```diff
@@ -329,6 +329,7 @@ public API getApi(String version, String context) {
                     api.setPolicy(resultSet.getString("API_TIER"));
                     api.setVersion(resultSet.getString("API_VERSION"));
                     api.setContext(resultSet.getString("CONTEXT"));
+                    api.setIsDefaultVersion(resultSet.getBoolean("PUBLISHED_DEFAULT_API_VERSION"));
```

Comment:
This logic is not right. "PUBLISHED_DEFAULT_API_VERSION" colum gives you the default version. What you could do is check "PUBLISHED_DEFAULT_API_VERSION" is equal to "API_VERSION" and set the boolean as true if equals

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9076/commits/9c76d41950a114213dcd434e71f496deec16f689